### PR TITLE
services/horizon: Add some manual-close improvements to integration tests

### PR DESCRIPTION
**Pull request by 2opremio**
* Poll Core to close ledgers synchronously
* Wait for manual-closing goroutine when submitting transactions

### CODE REVIEW

File: services/horizon/internal/test/integration/integration.go [hunk 1]
```diff
@@ -703,19 +720,26 @@ func (i *Test) SubmitTransaction(tx *txnbuild.Transaction) (proto.Transaction, e
...
+		// instrumenting Horizon to tell us when the submission is done.
 		time.Sleep(time.Millisecond * 100)
 		if err := i.CloseCoreLedger(); err != nil {
 			log.Fatalf("failed to CloseCoreLedger(): %s", err)
 		}
 	}()
-
-	return i.Client().SubmitTransactionXDR(txb64)
+	tx, err := i.Client().SubmitTransactionXDR(txb64)
+	wg.Wait()
```
Comment (bartekn) [inline services/horizon/internal/test/integration/integration.go:741]: I think it won't work if go routine closes the ledger before we actually submit transaction to core. I think we need to do it as explained in: https://github.com/stellar/go/pull/3167#discussion_r513376051.

Comment (2opremio) [reply services/horizon/internal/test/integration/integration.go:741]: I think that it's enough to manual-close in a loop until `SubmitTransactionXDR` returns.

Comment (bartekn) [reply services/horizon/internal/test/integration/integration.go:741]: This would work but there is no loop in go routine above.

Comment (2opremio) [reply services/horizon/internal/test/integration/integration.go:741]: Yep, I will add it.

**Comment by 2opremio** [comment]
@bartekn PTAL

File: services/horizon/internal/test/integration/integration.go (hunk 1 above)
Comment (2opremio) [reply services/horizon/internal/test/integration/integration.go:755]: Done

**Review by bartekn — APPROVED** [review]

**Pull request merged**